### PR TITLE
Remove dataclass emulation

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
towards https://github.com/lcompilers/lpython/issues/2276, https://github.com/lcompilers/lpython/issues/2274

### CODE REVIEW

**Comment by Shaikh-Ubaid** [comment]
The old `4` commits are part of the PR https://github.com/lcompilers/lpython/pull/2284. This PR needs a rebase.

**Comment by Shaikh-Ubaid** [comment]
I think this PR would hopefully pass after rebasing on top of https://github.com/lcompilers/lpython/pull/2284.

File: src/runtime/lpython/lpython.py [hunk 1]
```diff
@@ -48,13 +48,6 @@ def __getitem__(self, params):
     def __call__(self, arg):
         return self._convert(arg)
 
-def dataclass(arg):
-    def __class_getitem__(key):
-        return Array(arg, key)
-    arg.__class_getitem__ = __class_getitem__
-
-    return py_dataclass(arg)
-
```
Comment (certik) [inline src/runtime/lpython/lpython.py:57]: Beautiful! That's exactly what we need.

**Review by certik — APPROVED** [review]
I think this works! Thanks, great job.

**Pull request merged**